linode/manager
# change: [M3-8659] - Incorporate Product Family Groups in Side Nav

## Description 📝
More products are getting added to Cloud Manager, so we need to make sure our side navigation menu can accommodate them.

This PR introduces collapsible Product families into the side nav to allow more products to be shown while also giving users the ability to hide/show relevant families. Additionally, individual product icons have been removed in favor of Product family icons

Note: The Monitor product family icon is still in discussion. Create menu dropdown will be updated in M3-8728.

## Changes  🔄
- Product nav links have been moved inside an accordion except for Managed
  - The default state for the accordions is expanded
  - Collapsed accordions are saved to preferences so state is maintained
- Individual product icons have been removed, only product families will have icons
  - Empty state landing pages have been updated accordingly
- When the side menu is collapsed, only product family icons are displayed
  - Hovering over the side menu while it is in the collapsed state will expand the menu with all the items
- The active product family icon will be green

Code clean up:
- Removed old unused prefetch code 🧹 
- Migrated styles to Styled components
- Moved PrimaryLink to its own component

## Preview 📷
| Before  | After   |
| ------- | ------- |
| ![image](https://github.com/user-attachments/assets/3606c3a5-cbcc-4ee3-abc2-717c83140361) | <video src="https://github.com/user-attachments/assets/45607aee-b12d-4859-88cc-c41c6265aba7" /> |

## How to test 🧪
### Verification steps
(How to verify changes)
- Pull this PR locally and check different states of the side nav menu
- To test Managed, you can use the MSW to set Managed to enabled

## As an Author I have considered 🤔

*Check all that apply*
